dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/DefaultTrackerObjectsDeletionService.java

```diff
@@ -110,6 +111,17 @@ public TrackerTypeReport deleteEnrollments( TrackerBundle bundle, TrackerType tr
 
             ProgramInstance programInstance = programInstanceService.getProgramInstance( uid );
 
+            if ( programInstance == null )
```

Comment:
That shouldn't be here.
All the validations must happen in the validation phase.

Reply:
it should. but I guess its not currently checking enrollment existence

Reply:
resolved by reversing tracker type object. Now order of deletion is